src/updatechecker.cpp

```diff
@@ -219,6 +221,39 @@ UpdateChecker::UpdateChecker(): Thread("WinSparkle updates check")
 {
 }
 
+bool UpdateChecker::CreateInstallerProcess(const std::wstring filePath, DWORD dwCreationFlags, LPPROCESS_INFORMATION lpProcessInformation)
+{
+    STARTUPINFO             Startup;
+    PROCESS_INFORMATION     processInfo;
+
+    if (filePath == L"")
+    {
+        return false;
+    }
+
+    GetStartupInfo(&Startup);
+    Startup.wShowWindow = SW_SHOWNORMAL;
+    if (lpProcessInformation == NULL)
+    {
+        lpProcessInformation = &processInfo;
+    }
+
+    std::wstring fullCommandLine = filePath + L" -s";
+
+    if (!CreateProcess(NULL, const_cast<WCHAR*>(fullCommandLine.c_str()), NULL, NULL, TRUE, DETACHED_PROCESS, NULL, NULL, &Startup, lpProcessInformation))
+    {
+        return false;
+    }
+
+    if (lpProcessInformation == &processInfo)
+    {
+        CloseHandle(processInfo.hProcess);
+        CloseHandle(processInfo.hThread);
```

Comment:
Is the caller expected to do CloseHandle() on `lpProcessInformation.hProcess` and `lpProcessInformation.hThread` after `true` is returned? Isn't it too risky to allow/require the caller to do that? Is that how it is going to be used? Why does the caller need to have the Process Info after completion? Or the other way around, if the caller does NOT pass in a Process Info pointer, why can't we just return with an error and not worry about the local Process Info?

Reply:
I would tailor it for your exact use case, i.e. either make lpProcessInformation a mandatory not NULL parameter OR I would remove it altogether.

Reply:
Yes, I removed that parameter altogether.